livespark-form-modeler/livespark-form-modeler-codegen/livespark-form-modeler-codegen-impl/src/test/java/org/livespark/formmodeler/codegen/flow/impl/RoasterFlowProducerSourceGeneratorTest.java

```diff
@@ -0,0 +1,188 @@
+    private final String FULL_CLASS_NAME = "public class myFormNameFlowProducer extends FlowProducer<myFormName, myFormNameFormModel, myFormNameFormView, myFormNameListView, myFormNameRestService>";
+
+    private final int PACKAGE_INDEX = 8;
+    private final List<String> ABSTRACT_METHODS = Arrays.asList("modelToFormModel", "formModelToModel", "newModel", "getModelType", "getFormModelType");
+    private final List<String> ABSTRACT_METHODS_IMPLEMENTATIONS = Arrays
+            .asList("return new myFormNameFormModel(model)",
+                    "return formModel.getmyFormName()",
+                    "return new myFormName()",
+                    "return myFormName.class",
+                    "return myFormNameFormModel.class");
+    private final List<String> PRODUCER_METHODS = Arrays.asList("entityType", "create", "crud", "createAndReview", "view");
+    private final List<String> PRODUCER_METHODS_IMPLEMENTATIONS = Arrays
+            .asList("return myFormName.class",
+                    "return super.create()",
+                    "return super.crud()",
+                    "return super.createAndReview()",
+                    "return super.view()");
+
+    private final List<String> IMPORTS = Arrays.asList("org.juraj.mypackage.myFormName", "org.juraj.mypackage.myFormNameFormModel", "org.juraj.mypackage.myFormNameRestService", "org.livespark.flow.api.Unit", "java.util.Optional");
+
+    private int gslen;
+    private String generatedSource;
+    private RoasterFlowProducerSourceGenerator sourceGenerator;
+    private FormDefinition formDefinition;
+    private Path path;
+    private org.guvnor.common.services.project.model.Package root;
+    private org.guvnor.common.services.project.model.Package local;
+    private org.guvnor.common.services.project.model.Package shared;
+    private List<FormDefinition> projectForms;
+    private SourceGenerationContext context;
+
+    @Before
+    public void setUp() {
+        sourceGenerator = new RoasterFlowProducerSourceGenerator();
+
+        formDefinition = new FormDefinition();
+        formDefinition.setId(FORM_ID);
+        formDefinition.setName(FORM_NAME);
+
+        formDefinition.setModel(new DataObjectFormModel("employee", "org.juraj.mypackage.Employee"));
+
+        path = PathFactory.newPath(FORM_NAME, FORM_URI);
+        shared = new org.guvnor.common.services.project.model.Package(path, null, null, null, null, PACKAGE, "caption", "relativeCaption");
+        local = new org.guvnor.common.services.project.model.Package(path, null, null, null, null, PACKAGE, "caption", "relativeCaption");
+
+        context = new SourceGenerationContext(formDefinition, path, root, local, shared, null, projectForms);
+        generatedSource = sourceGenerator.generateJavaSource(context);
+        gslen = generatedSource.length();
+    }
+
+    @Test
+    public void generateJavaSource() {
+        assertThat(getImports()).containsAll(IMPORTS);
```

Comment:
Shouldn't this method generate the source as well? After all, that's what you're testing here.